cli/azd/cmd/middleware/middleware.go

```diff
@@ -48,17 +48,15 @@ type NextFn func(ctx context.Context) (*actions.ActionResult, error)
 // Middleware runner stores middleware registrations and orchestrates the
 // invocation of middleware components and actions.
 type MiddlewareRunner struct {
-	chain       []string
-	container   *ioc.NestedContainer
-	actionCache map[actions.Action]*actions.ActionResult
```

Comment:
Happy to other suggestions. Could not rerun the `package` action without removing the cache.

Since:
1. This cache was newly introduced
2. We already have caching in the service layer
3. No workflows actually exist that take advantage of caching currently

I removed the behavior. I could be wrong on any of the above assumptions, and happy to take other suggestions.